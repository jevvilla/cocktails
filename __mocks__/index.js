export const drink = {
  details: [
    {
      dateModified: '2016-08-31 19:48:22',
      idDrink: '16108',
      strAlcoholic: 'Alcoholic',
      strCategory: 'Cocktail',
      strCreativeCommonsConfirmed: 'No',
      strDrink: '9 1/2 Weeks',
      strDrinkAlternate: null,
      strDrinkDE: null,
      strDrinkES: null,
      strDrinkFR: null,
      strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg',
      'strDrinkZH-HANS': null,
      'strDrinkZH-HANT': null,
      strGlass: 'Cocktail glass',
      strIBA: null,
      strIngredient1: 'Absolut Citron',
      strIngredient2: 'Orange Curacao',
      strIngredient3: 'Strawberry liqueur',
      strIngredient4: 'Orange juice',
      strIngredient5: null,
      strIngredient6: null,
      strIngredient7: null,
      strIngredient8: null,
      strIngredient9: null,
      strIngredient10: null,
      strIngredient11: null,
      strIngredient12: null,
      strIngredient13: null,
      strIngredient14: null,
      strIngredient15: null,
      strInstructions:
        'Combine all ingredients in glass mixer. Chill and strain into Cocktail glass. Garnish with sliced strawberry.',
      strInstructionsDE:
        'Alle Zutaten in einem Glasmischer vermengen. Abk√ºhlen lassen und in das Cocktailglas abseihen. Mit Erdbeerscheiben garnieren.',
      strInstructionsES: null,
      strInstructionsFR: null,
      'strInstructionsZH-HANS': null,
      'strInstructionsZH-HANT': null,
      strMeasure1: '2 oz ',
      strMeasure2: '1/2 oz ',
      strMeasure3: '1 splash ',
      strMeasure4: '1 oz ',
      strMeasure5: null,
      strMeasure6: null,
      strMeasure7: null,
      strMeasure8: null,
      strMeasure9: null,
      strMeasure10: null,
      strMeasure11: null,
      strMeasure12: null,
      strMeasure13: null,
      strMeasure14: null,
      strMeasure15: null,
      strTags: null,
      strVideo: null,
    },
  ],
  idDrink: '16108',
  strDrink: '9 1/2 Weeks',
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg',
};

export const filteredIngredients = [
  'Absolut Citron',
  'Orange Curacao',
  'Strawberry liqueur',
  'Orange juice',
];

export const filteredMeasures = ['2 oz ', '1/2 oz ', '1 splash ', '1 oz '];

export const measuresAndIngredients = [
  '2 oz - Absolut Citron',
  '1/2 oz - Orange Curacao',
  '1 splash - Strawberry liqueur',
  '1 oz - Orange juice',
];
