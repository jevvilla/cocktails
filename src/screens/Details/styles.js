import { StyleSheet } from 'react-native';

import colors from '../../common/colors';
import styles from '../../common/styles';

export default StyleSheet.create({
  container: {
    ...styles.paddingVertical,
    ...styles.paddingHorizontal,
    backgroundColor: colors.background,
    flex: 1,
  },
  headerLeft: {
    ...styles.paddingHorizontal,
  },
  headerTitle: {
    ...styles.headerTitle,
  },
});
