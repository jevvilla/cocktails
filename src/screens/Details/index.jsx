import React, { PureComponent } from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import styles from './styles';
import colors from '../../common/colors';

class Details extends PureComponent {
  static navigationOptions = ({ navigation }) => ({
    headerLeft: (
      <TouchableOpacity style={styles.headerLeft} onPress={() => navigation.goBack(null)}>
        <Icon name="ios-arrow-round-back" size={45} color={colors.white} />
      </TouchableOpacity>
    ),
  });

  render() {
    return (
      <View style={styles.container}>
        <Text>Hi there! I am Details</Text>
      </View>
    );
  }
}

export default Details;
