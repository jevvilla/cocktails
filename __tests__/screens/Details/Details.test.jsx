import React from 'react';
import renderer from 'react-test-renderer';

import Details from '../../../src/screens/Details';
import { measuresAndIngredients } from '../../../__mocks__';

const params = {
  uri: 'headerTitle',
  measures: measuresAndIngredients,
  instructions: 'instructions',
};

const navigation = {
  getParam: jest.fn(key => params[key]),
  goBack: jest.fn(),
  navigate: jest.fn(),
  state: {
    params,
  },
  setParams: jest.fn(),
};

describe('Details', () => {
  const tree = renderer.create(<Details navigation={navigation} />);

  test('should match the snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  describe('navigationOptions', () => {
    const navigationOptions = Details.navigationOptions({ navigation });

    test('navigationOptions match snapshot', () => {
      expect(navigationOptions).toMatchSnapshot();
    });

    test('headerLeft IconButton onPress', () => {
      renderer.create(navigationOptions.headerLeft).root.props.onPress();

      expect(navigation.goBack).toBeCalledWith(null);
    });
  });
});
