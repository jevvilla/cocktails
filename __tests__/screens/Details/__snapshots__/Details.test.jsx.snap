// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details navigationOptions navigationOptions match snapshot 1`] = `
Object {
  "headerLeft": <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "paddingHorizontal": 16,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#fff"
      name="ios-arrow-round-back"
      size={35}
    />
  </TouchableOpacity>,
  "headerTitle": <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 22,
        "opacity": 0.8,
      }
    }
  />,
}
`;

exports[`Details should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#41AFC7",
      "flex": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "height": 650,
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1.3,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "headerTitle",
          }
        }
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        2 oz - Absolut Citron
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        1/2 oz - Orange Curacao
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        1 splash - Strawberry liqueur
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        1 oz - Orange juice
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 16,
              "opacity": 0.4,
            },
            Object {
              "paddingVertical": 16,
            },
          ]
        }
      >
        â€¢ How to prepate
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        instructions
      </Text>
    </View>
  </View>
</View>
`;
